### PR TITLE
Move delete saved trips functionality within edit functionality

### PR DESCRIPTION
**Pull request by AdrianaCeric**
**Description:**
Previously, the delete button on a saved trips card was seperate from the edit button, but it makes more sense if it were part of the edit page

**PR Checklist:**
- [x] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [x] Are all languages supported (Internationalization/Localization)?
- [x] Are appropriate Typescript types implemented?

image

### CODE REVIEW

File: lib/components/user/form-navigation-buttons.tsx [hunk 1]
```diff
@@ -15,19 +15,24 @@ const RightButton = styled(Button)`
...
+export interface ButtonType {
+  /** The component to use for the button. */
+  ButtonComponent?: ReactNode
+  /** Whether the button is disabled. */
   disabled?: boolean
   /** Triggered when the button is clicked. */
   onClick?: MouseEventHandler<Button>
   /** The text to display on the button (JSX elements accepted). */
-  text: ReactElement
+  text?: ReactElement
```
Comment (miles-grant-ibigroup) [inline lib/components/user/form-navigation-buttons.tsx:26]: Why is this optional?

File: lib/components/user/form-navigation-buttons.tsx [hunk 2]
```diff
@@ -15,19 +15,24 @@ const RightButton = styled(Button)`
   float: right;
 `
 
-interface ButtonType {
+export interface ButtonType {
+  /** The component to use for the button. */
+  ButtonComponent?: ReactNode
```
Comment (miles-grant-ibigroup) [inline lib/components/user/form-navigation-buttons.tsx:20]: Can we use `children` instead?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
Looks good! Just a few nitpicks

File: lib/components/user/form-navigation-buttons.tsx (hunk 1 above)
Comment (AdrianaCeric) [reply lib/components/user/form-navigation-buttons.tsx:26]: this button text was already defined in our separate `DeleteForm` component with i18n, so we didn't need it for this button and then we decided to make it optional

File: lib/components/user/form-navigation-buttons.tsx (hunk 2 above)
Comment (binh-dam-ibigroup) [reply lib/components/user/form-navigation-buttons.tsx:20]: Or `content`?

Comment (AdrianaCeric) [reply lib/components/user/form-navigation-buttons.tsx:20]: Good suggestions I think either make sense. I'll change to `content`

Comment (miles-grant-ibigroup) [reply lib/components/user/form-navigation-buttons.tsx]: No I meant using `children` so you don't have to pass a prop at all and can instead just use the child of the component!

File: lib/components/user/form-navigation-buttons.tsx (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/components/user/form-navigation-buttons.tsx:26]: Hm I'm not sure I love that... Can we name it `textOverride` or something?

**Comment by AdrianaCeric** [comment]
`sequential-pane-display` is being removed??

File: lib/components/user/form-navigation-buttons.tsx [hunk 3]
```diff
@@ -23,7 +23,7 @@ export interface ButtonType {
   /** Triggered when the button is clicked. */
   onClick?: MouseEventHandler<Button>
   /** The text to display on the button (JSX elements accepted). */
-  text?: ReactElement
+  textOverride?: ReactElement
```
Comment (binh-dam-ibigroup) [inline lib/components/user/form-navigation-buttons.tsx:26]: Not sure why you'd want to rename this field, as there is no default text that is set if this attribute is not defined.

Comment (miles-grant-ibigroup) [reply lib/components/user/form-navigation-buttons.tsx:26]: I thought @AdrianaCeric said that there was a default?

Comment (AdrianaCeric) [reply lib/components/user/form-navigation-buttons.tsx]: I think naming can go either way. what matters is functionality. We can rename and refactor some things in FormNavigationButtons but for now its circling around the point of this PR

File: lib/components/user/form-navigation-buttons.tsx (hunk 2 above)
Comment (AdrianaCeric) [reply lib/components/user/form-navigation-buttons.tsx]: I understood how to implement this only halfway, but having trouble subsequently rendering children of extraButton in SavedTripEditor

**Review by miles-grant-ibigroup — APPROVED** [review]
With all our conversation noted here, I think this is good enough for now!

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**